Hide the transaction details refund menu for non-refundable disputes

Jinksi opened the pull request:
Fixes #7960

### Changes proposed in this Pull Request

This PR hides the refund menu on the transaction details screen when it has a non-refundable dispute.

It introduces a new dispute utility function `isRefundable()` to determine if a dispute status is eligible or not. Dispute statuses that are refundable and should show the refund dropdown: `won`, `warning_needs_response`, `warning_under_review`, `warning_closed`.

https://github.com/Automattic/woocommerce-payments/blob/b864c62b4745919980f364bc459856e41096d7ff/client/disputes/utils.ts#L75-L78

**Transaction with a dispute not eligible for refund – refund menu is not rendered**

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/2bfab240-e2e9-4c51-8c91-2210cd508080)


**Transaction with a dispute eligible for refund – refund menu is rendered**

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/bdc59a07-17b9-48f3-9368-ee18d82ecfc0)


### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Ensure the refund menu does not appear on the transaction details screen:**
* Active or lost disputes – as a shopper, use the card `4000000000000259` at checkout.

**Ensure the refund menu appears on the transaction details screen:**
* Any non-disputed transactions.
* Inquiries – as a shopper, use the card `4000000000001976` at checkout.
* "Won" disputes – as a merchant, provide winning evidence by submitting the challenge form with `winning_evidence` in the “Additional details” field.


-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Jinksi committed:
Add dispute isRefundable util function

Jinksi committed:
Add changelog entry

Jinksi committed:
Only show transaction refund menu if dispute is refundable

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7962` or branch name `fix/7960-transaction-refund-eligible-disputes-only` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7960-transaction-refund-eligible-disputes-only) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7b47bf0b90674586693ae127c0acee04849b13ee
- Build time: 2024-01-04 22:39:09 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Add tests for dispute refund menu cases

Jinksi commented on client/payment-details/summary/test/index.test.tsx:
```diff
@@ -427,6 +427,13 @@ describe( 'PaymentDetailsSummary', () => {
+		// Refund menu is not rendered
+		expect(
+			screen.queryByRole( 'button', {
+				name: /Translation actions/i,
```
Note that the label for the refund menu button is `Translation actions` – I've opened #7964 to correct this label.

Jinksi commented:
Requesting an optional review from @mgascam @KarlisJ since you were mentioned in the issue thread and have a good understanding of the new refund menu flow.

kalessil reviewed (approved):
LGTM (code review only).

KarlisJ reviewed (approved):
Thank you @Jinksi for handling this!

Functionally the changes are looking good, hence I'm approving the PR.

However, as Vlad already mentioned there's a slight misconception that the menu is meant only for refund related actions. 

While it's true for now, the idea is that we'll be able to add more transaction related actions to the menu in the future. I didn't want to over-engineer things initially but now seeing that that some confusion is already there, I'm re-thinking that decision. 

Ideally, we should have extract an array for refund actions and show the menu only if any action is available. Something like this:

```JS
const refundActions = [];
// populate refund actions based on conditions.
// this is where we would add more actions in the future.
const controlActions = [...refundActions];
const showControlMenu = !! refundActions.length;
```

This way it's clear for the future selves how to add more actions to the controls and the visibility of controls menu will be handled automatically. And we we'll be able to keep the logic in single place out from the template.

Jinksi committed:
Merge branch 'develop' into fix/7960-transaction-refund-eligible-disputes-only

Jinksi committed:
Rename var to `showControlMenu` to indicate refund menu may include other actions in the future

Jinksi commented:
> we'll be able to add more transaction related actions to the menu in the future

Ah, yes I did assume this was for refunds only. I think your initial approach was correct – it's OK for the small refactor to occur if or when new transaction actions are added.

I've renamed the var `showRefundMenu` to `showControlMenu` and added a comment indicating that the menu may include other actions in the future in d767dc1.

KarlisJ commented:
@Jinksi Thank you for addressing this. I like the simplicity of your last commit. You address the confusion without any over-engineering.

Jinksi committed:
Merge branch 'develop' into fix/7960-transaction-refund-eligible-disputes-only

Jinksi commented:
Failing shopper e2e tests are unrelated (see failing tests on all PRs https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-pull-request.yml).

Adding to merge queue
